packages/flutter/lib/src/rendering/object.dart

```diff
@@ -4583,133 +4328,609 @@ mixin RelayoutWhenSystemFontsChangeMixin on RenderObject {
-  /// previously painted [RenderObject]s unreachable for accessibility purposes.
-  ///
-  /// See also:
-  ///
-  ///  * [SemanticsConfiguration.isBlockingSemanticsOfPreviouslyPaintedNodes]
-  ///    describes what semantics are dropped in more detail.
-  final bool dropsSemanticsOfPreviousSiblings;
-
-  /// Returns [_InterestingSemanticsFragment] describing the actual semantic
-  /// information that this fragment wants to add to the parent.
-  List<_InterestingSemanticsFragment> get mergeUpFragments;
+/// This is used by `_SemanticsFragment` to access child fragments. Using this
+/// class provides a way for child render object to provide latest
+/// `_InterestingSemanticsFragment` in case of rebuilds. Instead of caching the
+/// child fragments directly, caching the child's
+/// _InterestingSemanticsFragmentProvider avoids the need to update all parent
+/// fragment's cache when a child render object rebuild.
+abstract class _InterestingSemanticsFragmentProvider {
+  /// Gets the latest fragment.
+  _InterestingSemanticsFragment getFragment();
 }
 
-/// A container used when a [RenderObject] wants to add multiple independent
-/// [_InterestingSemanticsFragment] to its parent.
-///
-/// The [_InterestingSemanticsFragment] to be added to the parent can be
-/// obtained via [mergeUpFragments].
-class _ContainerSemanticsFragment extends _SemanticsFragment {
-  _ContainerSemanticsFragment({
-    required super.dropsSemanticsOfPreviousSiblings,
-    required this.siblingMergeGroups,
-  });
-
-  final List<List<_InterestingSemanticsFragment>> siblingMergeGroups;
+class _IncompleteSemanticsFragmentProvider extends _InterestingSemanticsFragmentProvider {
+  _IncompleteSemanticsFragmentProvider(this.fragment);
 
-  @override
-  void addAll(Iterable<_InterestingSemanticsFragment> fragments) {
-    mergeUpFragments.addAll(fragments);
-  }
+  final _IncompleteSemanticsFragment fragment;
 
   @override
-  final List<_InterestingSemanticsFragment> mergeUpFragments = <_InterestingSemanticsFragment>[];
+  _InterestingSemanticsFragment getFragment() => fragment;
 }
 
-/// A [_SemanticsFragment] that describes which concrete semantic information
-/// a [RenderObject] wants to add to the [SemanticsNode] of its parent.
-///
-/// Specifically, it describes which children (as returned by [compileChildren])
-/// should be added to the parent's [SemanticsNode] and which [config] should be
-/// merged into the parent's [SemanticsNode].
-abstract class _InterestingSemanticsFragment extends _SemanticsFragment {
-  _InterestingSemanticsFragment({
-    required RenderObject owner,
-    required super.dropsSemanticsOfPreviousSiblings,
-  }) : _ancestorChain = <RenderObject>[owner];
-
-  /// The [RenderObject] that owns this fragment (and any new [SemanticsNode]
-  /// introduced by it).
-  RenderObject get owner => _ancestorChain.first;
+/// A wrapper class for a [RenderObject] that provides semantics related
+/// properties and compilations.
```

Comment:
Can you update this comment to be a little more informative? It should roughly lay out the lifecycle of a semantics update and describe when a user of this object (i.e. a RenderObject) should call which method.